### ISSUE TITLE
check if we a re not doing softmwax twice for canned models

### ISSUE THREAD

**Issue by undertherain** [undertherain/benchmarker#62]
we add cross entropy loss layer explicitly in a wrapper
but we should check if modes such as resnet50 from torchvision module don't come with softwmax output already emedded.
This might be the case as Jens notices that loss goes down slwer

**Comment by undertherain** [comment]
the other reason for loss decrease to slow down might be that we stopped used patterned synthetic data and just using random tensors instead

**Comment by vatai** [comment]
I've checked the models. They have `fc=Linear(...)` as its last layer (without softmax) - AFAIK.
```
In [1]: import torchvision.models as models                                                                        

In [2]: net = models.resnet50()                                                                                    

In [3]: print(net)                                                                                                 
ResNet(
  (conv1): Conv2d(3, 64, kernel_size=(7, 7), stride=(2, 2), padding=(3, 3), bias=False)
  # ... snip ...
  )
  (avgpool): AdaptiveAvgPool2d(output_size=(1, 1))
  (fc): Linear(in_features=2048, out_features=1000, bias=True)
)
```

**Comment by vatai** [comment]
From Jens:

old benchmarker:
```
Train Epoch: 1 [0/1 (0%)]	Loss: 6.407102
Train Epoch: 2 [0/1 (0%)]	Loss: 4.219720
Train Epoch: 3 [0/1 (0%)]	Loss: 0.665021
Train Epoch: 4 [0/1 (0%)]	Loss: 0.005704
Train Epoch: 5 [0/1 (0%)]	Loss: 0.000027
Train Epoch: 6 [0/1 (0%)]	Loss: 0.000000
```
new benchmarker:
```
Train Epoch: 1 [0/1 (0%)]	Loss: 7.102377 Time: 3.758516
Train Epoch: 2 [0/1 (0%)]	Loss: 6.960844 Time: 3.441289
Train Epoch: 3 [0/1 (0%)]	Loss: 6.963915 Time: 3.575963
Train Epoch: 4 [0/1 (0%)]	Loss: 6.927278 Time: 3.563911
Train Epoch: 5 [0/1 (0%)]	Loss: 6.748653 Time: 3.432570
Train Epoch: 6 [0/1 (0%)]	Loss: 6.599795 Time: 3.564205
```
Jens' remark: ignore the time thingy in the end, that was my debugging info

Old Benchmarker commit hash: c793220e86fcee2d06548e61d021227842c85000

**Comment by vatai** [comment]
Commit 7743fd9f4f18b3981fc105949f80fd1705e607ae "brakes" this behaviour.

**Comment by vatai** [comment]
Ok, I think I've found it. This is my **guess** what caused the change. Will verify in a sec.
```diff
diff --git a/benchmarker/util/data/synthetic/helpers.py b/benchmarker/util/data/synthetic/helpers.py
index e067ec8..801a207 100644
--- a/benchmarker/util/data/synthetic/helpers.py
+++ b/benchmarker/util/data/synthetic/helpers.py

@@ -38,5 +39,5 @@ def gen_classification_data(params, height, width=None):
     shape = shape + params["problem"]["size"][1:]
 
     X = np.random.random(shape).astype(np.float32)
-    Y = np.random.random(shape[:2]).astype(np.int64)
+    Y = np.random.randint(0, num_cls, shape[:2])
     return X, Y
```

**Comment by vatai** [comment]
# Sanity checked and verified.

So the above line caused the difference in the output.

## Check 1
Checked out `master` (commit e20f59d9c2326367eb7ce329b5a251e2ce61d1eb) and tried switching between the two versions of the line mentioned in the comment above, and it resulted in the expected change.

## Check 2
Looked at the execution time - it's almost identical.
